gravitee-am-service/src/main/java/io/gravitee/am/service/impl/DeviceServiceImpl.java

```diff
@@ -110,6 +111,6 @@ public Completable delete(String domain, UserId user, String deviceId, User prin
                             String.format("An error occurs while trying to delete factor: %s", deviceId), ex));
                 })
                 .doOnComplete(() -> auditService.report(AuditBuilder.builder(UserAuditBuilder.class).principal(principal).type(EventType.DEVICE_DELETED)))
```

Comment:
looks like the reference is missing from here also and we are not specifying the deviceId (as target)